src/preact/base-element.js

```diff
@@ -209,7 +209,20 @@ export class PreactBaseElement extends AMP.BaseElement {
   isLayoutSupported(layout) {
     const Ctor = this.constructor;
     if (Ctor['layoutSizeDefined']) {
-      return isLayoutSizeDefined(layout);
+      return (
+        isLayoutSizeDefined(layout) ||
```

Comment:
`isLayoutSizeDefined` only excludes `CONTAINER` and `NODISPLAY` among all layout types. Couldn't this just be `return true`?

Reply:
I think so. But since layouts are enumerated, I think this might be dangerous. To clarify, I think using "container" layout for elements without layout was not a great idea from the get go. We should have called it `layout=auto` and allowed different elements to interpret it as they see fit. So, here I'm now trying to be very explicit that we sort of assume "container" to be this "auto" layout.

Reply:
To clarify this a bit more. In AMP world, `layout=container` means `contain: content`. But `layoutSizeDefined` components mean `contain: strict`. So "container" is used here only because that's the layout AMP assigns to these components by default. So I wanted to be very explicit about this nuance.

Reply:
I see, thanks for clarifying.